api/apiclient.go

```diff
@@ -270,6 +270,25 @@ func open(
 	return st, nil
 }
 
+type ConnectionForModelFunc func(*Info) (func(string) (Connection, error), error)
+
+// APIConnForModelFunc provides a function which returns a model API
```

Comment:
also, maybe "NewConnectionForModelFunc" would be more appropriate? ConnectionForModel does not itself return a Connection.